### ISSUE TITLE
Query Logging: bringing back the execution time

### ISSUE THREAD

**Issue by Slamdunk** [doctrine/dbal#6242]
Hello, I've seen that https://github.com/doctrine/dbal/pull/4967 deprecated `DebugStack` in favour of a logging middleware.

While I totally agree with the issues reported in https://github.com/doctrine/dbal/issues/4941, for a debugging session it is a huge miss the absence of the exection time for each query.

In https://github.com/Slamdunk/dbal-debugstack-middleware I replicated the `DebugStack` behavior with a middleware (without the `psr/log`).

I wonder if you would accept a PR that edits https://github.com/doctrine/dbal/pull/4967 in order to log the queries **after** the execution, so we can also log the execution time.

**Comment by derrabus** [comment]
> I wonder if you would accept a PR that edits #4967 in order to log the queries **after** the execution, so we can also log the execution time.

I think logging the query first is on purpose and we don't want to change that.

I'm having trouble building a one-size-fits-all profiling solution. Which information should be gathered and how and where it should be stored depends on the application framework or profiling tool for which we gather that data. Symfony for instance has its own "debug middleware" which collects execution times for the Symfony profiler toolbar. You might want to look into Symfony's DoctrineBridge package for inspiration. Sentry also ships a Middleware for tracing queries. It does similar things, but with a different approach.

That library of yours looks great and if it works for you, please use it, advertise it, talk about it. 🙂 

tl;dr: There are already good solutions out there, leveraging the middleware system and tailoring profiling for their respective framework. I don't think we need a generic solution in DBAL's core.

**Comment by Slamdunk** [comment]
:+1:
